src/config/networks.ts

```diff
@@ -73,19 +73,28 @@ export const configExtender: ConfigExtender = (finalConfig, userConfig) => {
   const arapp = readArappIfExists()
   if (arapp && typeof arapp.environments === 'object') {
     for (const [networkName, network] of Object.entries(arapp.environments)) {
-      if (network.network && finalConfig.networks[network.network]) {
+      if (network.network && finalConfig.networks[networkName]) {
         const finalNetwork = finalConfig.networks[
-          network.network
+          networkName
         ] as HttpNetworkConfig
 
         // Append registry address
         if (network.registry) {
           finalNetwork.ensAddress = network.registry
         }
+      } else if (network.network && finalConfig.networks[network.network]) {
+        // Add missing network from arapp.json
+        const finalNetwork = Object.assign(
+          {},
+          finalConfig.networks[network.network]
+        ) as HttpNetworkConfig
```

Comment:
It appears we manipulate the object in the other case (`if (network.network && finalConfig.networks[networkName])`) as well, so perhaps we should clone the object in both cases?

Also wondering, with typescript, can we do:

```
const finalNetwork = { ...finalConfig.networks[network.network] } as HttpNetworkConfig
```

Reply:
Good points, I'll refactor a bit the code with the above suggestions.

About typescript, I looked and it should not be possible if we were using ES5 syntax because `Object.assign` is ES6. But because we are using `"target": "ES2017"` for the typescript compiler configuration, is not an issue.